src/main/java/com/openshift/internal/restclient/capability/resources/DockerRegistryImageStreamImportCapability.java

```diff
@@ -0,0 +1,277 @@
+			this.responseType = responseType;
+			this.data = data;
+		}
+		public Object getResponseType() {
+			return responseType;
+		}
+		public String getData() {
+			return data;
+		}
+	}
+
+	@Override
+	public IImageStreamImport importImageMetadata(DockerImageURI uri) {
+		
+		SslContextFactory sslFactory = new SslContextFactory(true);
+		HttpClient client = new HttpClient(sslFactory);
+		client.setConnectTimeout(TIMEOUT);
+		try {
+			client.start();
+			if(registryExists(client)) {
+				String token = null;
+				DockerResponse response = retrieveMetaData(client, token, uri);
+				if(DockerResponse.AUTH.equals(response.getResponseType())){
+					LOG.debug("Unauthorized.  Trying to retrieve token");
+					token = retrieveAuthToken(client, response.getData());
+					response = retrieveMetaData(client, token, uri);
+				}
+				if(DockerResponse.DATA.equals(response.getResponseType())) {
+					String meta = response.getData();
+					LOG.debug("Raw Docker image metadata: " + meta);
+					return buildResponse(meta, uri);
+				}else {
+					LOG.info("Unable to retrieve image metadata from docker registry");
+					return buildErrorResponse(uri);
+				}
+			}
+		} catch (Exception e) {
+			LOG.error("Exception while trying to retrieve image metadata from docker", e);
+		}finally {
+			try {
+				if(client != null) {
+					client.stop();
+				}
+			} catch (Exception e) {
+				LOG.warn("Exception while trying to stop http client", e);
+			}
+		}
+		return buildErrorResponse(uri);
+	}
+
+	private IImageStreamImport buildErrorResponse(DockerImageURI uri) {
+		ModelNodeBuilder builder = new ModelNodeBuilder()
+				.set(STATUS_STATUS, "Failure")
+				.set("status.message", String.format("you may not have access to the Docker image \"%s\"", uri.getUriWithoutHost()))
+				.set("status.reason", "Unauthorized")
+				.set("status.code", IHttpConstants.STATUS_UNAUTHORIZED);
+		
+		return buildImageStreamImport(uri, builder);
+	}
+
+	private IImageStreamImport buildResponse(String meta, DockerImageURI uri) {
+		ModelNode raw = ModelNode.fromJSONString(meta);
+		ModelNode last = findNewestHistoryEntry(raw);
+		ModelNode containerConfig = last.remove("container_config");
+		last.get("ContainerConfig").set(containerConfig);
+		
+		ModelNodeBuilder builder = new ModelNodeBuilder()
+			.set(STATUS_STATUS, "Success")
+			.set("tag", uri.getTag())
+			.set("image.metadata.name", uri.getName())
+			.set(ImageStreamImport.IMAGE_DOCKER_IMAGE_REFERENCE,uri.getUriUserNameAndName())
+			.set("image.dockerImageMetadata", last);
+
+		return buildImageStreamImport(uri, builder);
+	}
+	
+	private ImageStreamImport buildImageStreamImport(DockerImageURI uri, ModelNodeBuilder builder) {
+		ImageStreamImport isImport = (ImageStreamImport) factory.stub(ResourceKind.IMAGE_STREAM_IMPORT, uri.getName(), this.project.getName());
+		ModelNode root = isImport.getNode();
+		ModelNode images = JBossDmrExtentions.get(root,null, ImageStreamImport.STATUS_IMAGES);
+		images.add(builder.build());
```

Comment:
any specific reason to build.build() only down here? Why not pass the result of the builder as parameter?